### PR TITLE
usbdev: Fix build error with BOARD_USBDEV_SERIALSTR

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary

Fix build error and warning when enable BOARD_USBDEV_SERIALSTR.

## Impact

drivers/usbdev

## Testing

on Spresense board.

### CODE REVIEW

File: drivers/usbdev/composite.h [hunk 1]
```diff
@@ -50,6 +50,13 @@
 #define COMPOSITE_CONFIGIDNONE        (0)  /* Config ID = 0 means to return to address mode */
 #define COMPOSITE_CONFIGID            (1)  /* The only supported configuration ID */
 
+/* Descriptor strings */
+
+#define COMPOSITE_MANUFACTURERSTRID     (1)
```
Comment (xiaoxiang781216) [inline drivers/usbdev/composite.h:55]: these string is only used in drivers/usbdev/composite_desc.c, so don't need expose to public interface

File: drivers/usbdev/composite.c [hunk 2]
```diff
@@ -38,6 +38,10 @@
 #include <nuttx/usb/usbdev.h>
 #include <nuttx/usb/usbdev_trace.h>
 
+#ifdef CONFIG_BOARD_USBDEV_SERIALSTR
+#include <nuttx/board.h>
```
Comment (xiaoxiang781216) [inline drivers/usbdev/composite.c:42]: could you point out where composite driver use the interface from nuttx/board.h?

Comment (SPRESENSE) [reply drivers/usbdev/composite.c:42]: For calling `board_usbdev_serialstr()` at line 385

File: drivers/usbdev/composite.h (hunk 1 above)
Comment (SPRESENSE) [reply drivers/usbdev/composite.h:55]: I know, it is only use COMPOSITE_SERIALSTRID in composite.c actually, but they are the same string ID group defs, so I moved them all.

Comment (xiaoxiang781216) [reply drivers/usbdev/composite.h:55]: let's algin with other.lines.

Comment (SPRESENSE) [reply drivers/usbdev/composite.h:55]: here

https://github.com/apache/nuttx/blob/dfcf2dee293a7ac97d1a151ca82f3c802a7650ee/drivers/usbdev/composite.c#L393

Comment (xiaoxiang781216) [inline drivers/usbdev/composite.h:55]: My meaning is this:
```suggestion
#define COMPOSITE_MANUFACTURERSTRID   (1)
```

**Comment by SPRESENSE** [comment]
@xiaoxiang781216 

I fix it. Please review again.

File: drivers/usbdev/composite.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/usbdev/composite.c:42]: this line need intent too:
```suggestion
#  include <nuttx/board.h>
```

**Comment by SPRESENSE** [comment]
@xiaoxiang781216

I fix it. Please review again.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**